packages/common/src/types/coder.ts

```diff
@@ -71,6 +71,10 @@ export type DashboardTileAsCode = Omit<DashboardTile, 'properties' | 'uuid'> & {
         | DashboardLoomTileProperties['properties'];
 };
 
+export type DashboardTileWithSlug = DashboardTile & {
```

Comment:
why not adding an optional `tileSlug: string | undefined` to DashboardTile instead ?  

as long as we keep it optional so it is  backwards compatible, it should be fine to modify existing types.

Whatever you find more readable and scalable. this also works.